qoqo_qiskit/qoqo_qiskit/backend/backend.py

```diff
@@ -0,0 +1,264 @@
+        if (
+            not run_options["MeasurementInfo"]
+            and not run_options["SimulationInfo"]["PragmaGetStateVector"]
+            and not run_options["SimulationInfo"]["PragmaGetDensityMatrix"]
+        ):
+            raise ValueError(
+                "The Circuit does not contain Measurement, PragmaGetStateVector"
+                " or PragmaGetDensityMatrix operations. Simulation not possible."
+            )
+        #   - if both StateVector and DensityMatrix pragmas are involved
+        if (
+            run_options["SimulationInfo"]["PragmaGetStateVector"]
+            and run_options["SimulationInfo"]["PragmaGetDensityMatrix"]
+        ):
+            raise ValueError(
+                "The Circuit contains both a PragmaGetStateVector"
+                " and a PragmaGetDensityMatrix instruction. Simulation not possible."
+            )
+        #   - if more than 1 type of measurement is involved
+        if len(run_options["MeasurementInfo"]) > 1:
+            raise ValueError("Only input Circuits containing one type of measurement.")
+
+        # Handle simulation Options
+        shots = 200
+        custom_shots = 0
+        sim_type = "automatic"
+        if run_options["SimulationInfo"]["PragmaGetStateVector"]:
+            compiled_circuit.save_statevector()
+            sim_type = "statevector"
+        elif run_options["SimulationInfo"]["PragmaGetDensityMatrix"]:
+            compiled_circuit.save_density_matrix()
+            sim_type = "density_matrix"
+        if "PragmaRepeatedMeasurement" in run_options["MeasurementInfo"]:
+            for el in run_options["MeasurementInfo"]["PragmaRepeatedMeasurement"]:
+                if el[1] > custom_shots:
+                    custom_shots = el[1]
+        if "PragmaSetNumberOfMeasurements" in run_options["SimulationInfo"]:
+            for el in run_options["SimulationInfo"]["PragmaSetNumberOfMeasurements"]:
+                if el[1] > custom_shots:
+                    custom_shots = el[1]
+        if custom_shots != 0:
+            shots = custom_shots
+
+        # Simulation
+        result = self.qiskit_backend.run(
+            compiled_circuit, shots=shots, memory=True
+        ).result()
+
+        # Result transformation
+        if sim_type == "automatic":
+            transformed_counts = self._counts_to_registers(result.get_memory())
+            for reg in output_bit_register_dict:
+                output_bit_register_dict[reg] = transformed_counts.pop()
+        if sim_type == "statevector":
+            vector = list(np.asarray(result.data(0)["statevector"]).flatten())
+            for reg in output_complex_register_dict:
+                output_complex_register_dict[reg].append(vector)
+        if sim_type == "density_matrix":
+            vector = list(np.asarray(result.data(0)["density_matrix"]).flatten())
+            for reg in output_complex_register_dict:
+                output_complex_register_dict[reg].append(vector)
```

Comment:
Should this be an `if ...if else ... if else` loop or can these occur simultaneously?